### PR TITLE
load hive-site.xml for flink catalog

### PR DESCRIPTION
**Pull request by zhangjun0x01**
#1437  

In order to be compatible with various flink submission modes, FlinkCatalogFactory provides a variable hive-site-path to specify the path of hive-site.xml, which can be a local path or an hdfs path.

### CODE REVIEW

File: flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java [hunk 1]
```diff
@@ -121,4 +135,36 @@ protected Catalog createCatalog(String name, Map<String, String> properties, Con
   public static Configuration clusterHadoopConf() {
     return HadoopUtils.getHadoopConfiguration(GlobalConfiguration.loadConfiguration());
   }
+
+  private void loadHiveConf(Configuration configuration, Map<String, String> properties) {
+    String hiveConfPath = properties.get(HIVE_SITE_PATH);
+    Path path = new Path(hiveConfPath);
+    if (hiveConfPath.startsWith("hdfs")) {
```
Comment (holdenk) [inline flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java:142]: It seems weird to limit this to HDFS and not other hadoop compatible file systems, unless I'm missing something?

Comment (zhangjun0x01) [reply flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java:142]: yes ,you are right ,we should support more storage to store hive configuration file

Comment (kbendick) [reply flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java:142]: Could we also use a named constant for these string literals for the filesystem scheme? Maybe HDFS_SCHEME in this case would work.

File: flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java [hunk 2]
```diff
@@ -100,12 +108,18 @@ protected CatalogLoader createCatalogLoader(String name, Map<String, String> opt
...
+    properties.add(HIVE_SITE_PATH);
     return properties;
   }
 
   @Override
   public Catalog createCatalog(String name, Map<String, String> properties) {
-    return createCatalog(name, properties, clusterHadoopConf());
+    Configuration configuration = clusterHadoopConf();
+    String catalogType  = properties.get(ICEBERG_CATALOG_TYPE);
+    if (catalogType.equals("hive")) {
```
Comment (kbendick) [inline flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java:119]: Same note here about a named constant for this string literal.

File: flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java [hunk 3]
```diff
@@ -121,4 +135,36 @@ protected Catalog createCatalog(String name, Map<String, String> properties, Con
...
+                " please provide a hdfs path for hive-site.xml", hiveConfPath);
+      } else {
+        configuration.addResource(path);
+      }
+    }
+  }
+
+  private void download(Configuration configuration, Path hdfsHiveSitePath) {
+    try {
+      File tmpFile = File.createTempFile("hive-site.xml-", "");
```
Comment (kbendick) [inline flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java:157]: Is the temp file `hive-site.xml-` supposed to have that trailing dash? Is this possibly a typo?

Comment (zhangjun0x01) [reply flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java:157]: My initial idea was to add a prefix to distinguish it from the actual hive-site.xml. We can add a .tmp suffix to make users understand that it is only a temporary file

**Comment by rdblue** [comment]
@zhangjun0x01, looks like this is picking up changes from other PRs. Could you fix that, please?

File: flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java (hunk 3 above)
Comment (kbendick) [reply flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java]: That sounds much better to me. `.tmp` definitely better conveys the intent to me. Thanks for updating that!

File: flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java [hunk 4]
```diff
@@ -52,13 +56,21 @@
...
+  public static final String ICEBERG_CATALOG_TYPE_HIVE = "hive";
+  public static final String ICEBERG_CATALOG_TYPE_HADOOP = "hadoop";
+
   public static final String HIVE_URI = "uri";
   public static final String HIVE_CLIENT_POOL_SIZE = "clients";
   public static final String HADOOP_WAREHOUSE_LOCATION = "warehouse";
 
+  public static final String HIVE_SITE_PATH = "hive-site-path";
+  public static final String HIVE_SITE_SCHEMA_FILE = "file";
+  public static final String HIVE_SITE_SCHEMA_HDFS = "hdfs";
```
Comment (kbendick) [inline flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java:68]: Possible nit / typo / my own confusion on the above two: I think the term is `SCHEME` and not `SCHEMA`, but could be one of those instances where the term is overloaded. I've always heard of that part of the URI referred to as `scheme`.

Comment (kbendick) [reply flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java:68]: I see in your `getSchema` function that the call to the Hadoop fs API uses `scheme` so I think it is scheme.

```
String schema = path.toUri().getScheme();
```

**Pull request closed without merging**